### PR TITLE
ccxt feed: pass feed params and fix partial data

### PR DESCRIPTION
**Pull request by Dave-Vallance**
I discovered an issue where partial candle data was being downloaded from exchanges using the fetch OHLC method. Because the partial candle has a newer timestamp, it was added to the queue and processed as a full candle. The feed would then wait for the next timestamp which would also be a partial candle.

The fix is to check if the timestamp received is after the current UTC time. If so, it is from the future and not a complete candle. We can therefore skip adding it to the queue until it is complete

The other improvement is to allow passing params to the fetch OHLCV method. Some exchanges support some extra features which can be passed using `params

### CODE REVIEW

File: backtrader/feeds/ccxt.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ class CCXT(DataBase):
     params = (
         ('historical', False),  # only historical download
         ('backfill_start', False),  # do backfilling at the start
+        ('fetch_ohlcv_params', {})
```
Comment (bartosh) [inline backtrader/feeds/ccxt.py]: please, move this to another commit. This has nothing to do with fixing incomplete candles.

File: backtrader/feeds/ccxt.py [hunk 2]
```diff
@@ -121,16 +122,22 @@ def _fetch_ohlcv(self, fromdate=None):
...
+
                 if None in ohlcv:
                     continue
 
                 tstamp = ohlcv[0]
+
+                # Prevent from loading incomplete data
+                if tstamp > (time.time() * 1000):
+                    continue
+
```
Comment (bartosh) [inline backtrader/feeds/ccxt.py:137]: This looks like ccxt issue https://github.com/ccxt/ccxt/issues/2206. Should we ask ccxt guys to fix it?

File: backtrader/feeds/ccxt.py (hunk 1 above)
Comment (Dave-Vallance) [reply backtrader/feeds/ccxt.py]: Hi Ed,
Actually, I think this one is related somewhat. Some exchanges allow you to specify whether partial candles are provided. We can optionally pass these exchange parameters if/when needed using this.

File: backtrader/feeds/ccxt.py (hunk 2 above)
Comment (Dave-Vallance) [reply backtrader/feeds/ccxt.py:137]: It looks like that issue was assigned. So perhaps you can put this on hold until the fix is released and tested. 

I will continue to use this fix in my fork of yours as I happen to need it until the ccxt issue is fixed. 

BTW - Do you see this issue with the exchanges you use? I notice the ticket only mentions it affects "some" exchanges. Currently, I am testing on bitmex but that is only because it has a test server with paper trading.

Comment (bartosh) [reply backtrader/feeds/ccxt.py:137]: OK, let's put it on hold then. I might consider merging it if ccxt issue not fixed for a long time. Splitting this commit would also help.

I didn't see this issue before. I have PR #6 about it, but it also mentions bitmex.

**Comment by FiV0** [comment]
@Dave-Vallance Just wanted to clarify some things because you seem to be working on the same things.

1.) This [issue](https://github.com/ccxt/ccxt/issues/2206) has only partially something to do with partial data. It concerns the inconsistency that some exchanges use the ending time as time stamp and some the beginning time. So for example binance uses 16:59 as time stamp and bitmex 17:00 for the minute 16:59 - 17:00.

2.) It is therefore not enough just to check `if tstamp > (time.time() * 1000)` as in the case of `binance` you will get partial data for the current minute, but  `tstamp < (time.time() * 1000)`.

3.) If the ccxt guys decide to make the beginning time as time stamp for candles, we need to do more processing. Because if you feed at for example at 17:00 the **finished** `5m` candle 16:55-17:00 (time stamp 16:55) and the `1m` candle 16:59-17:00 (time stamp 16:59) to backtrader the `next()` will be called twice as the time stamps are not the same.

File: backtrader/feeds/ccxt.py (hunk 1 above)
Comment (bartosh) [reply backtrader/feeds/ccxt.py]: I don't see where is this parameter used in this PR. It's just added here and that's it.

BTW, ccxt feed has 'config' parameter. It's also passed to the __init__ of the ccxt store. Can it be used to provide ohlcv parameters?

**Comment by bartosh** [comment]
>It is therefore not enough just to check if tstamp > (time.time() * 1000) as in the case of binance you will get partial data for the current minute, but tstamp < (time.time() * 1000)

Would it make sense combine this PR with #6 somehow or extend it to make the fix more complete?
